### ISSUE TITLE
Crashes when opening images from other apps (Content Provider)

### ISSUE THREAD

**Issue by rugk** [UnevenSoftware/LeafPic#148]
I have an app which passes images using a [content provider](https://developer.android.com/guide/topics/providers/content-providers.html) (as a more secure method for sharing files). However when I use this app to share a image with LeafPic, LeafPic crashes.

It also failed when I instructed the other app to save my image and then used the "File saved" notification to pass the image to LeafPic.

Here some lines from logcat:

```
W/PackageManager(  621): Not granting permission android.permission.WRITE_EXTERNAL_STORAGE to package com.horaapps.leafpic because it was previously installed without
W/PackageManager(  621): Not granting permission android.permission.READ_EXTERNAL_STORAGE to package com.horaapps.leafpic because it was previously installed without
W/PackageManager(  621): Not granting permission android.permission.WRITE_EXTERNAL_STORAGE to package com.horaapps.leafpic because it was previously installed without
W/PackageManager(  621): Not granting permission android.permission.READ_EXTERNAL_STORAGE to package com.horaapps.leafpic because it was previously installed without
I/PackageManager(  621): Setting last chosen activity com.horaapps.leafpic/.PhotoPagerActivity for user 0:
I/ActivityManager(  621): START u0 {act=android.intent.action.VIEW dat=content://ch.*******.app.fileprovider/tmp_files/****EXAMPLEFILE****.gif typ=image/* flg=0x3000001 cmp=com.horaapps.leafpic/.PhotoPagerActivity} from uid 10202 on display 0
I/ActivityManager(  621): Start proc 7582:com.horaapps.leafpic/u0a249 for activity com.horaapps.leafpic/.PhotoPagerActivity
W/System  ( 7582): ClassLoader referenced unknown path: /data/app/com.horaapps.leafpic-1/lib/arm
W/XPrivacy( 7582): Hooking package=com.horaapps.leafpic
W/System.err( 7582):    at com.horaapps.leafpic.utils.StringUtils.quoteReplace(StringUtils.java:157)
W/System.err( 7582):    at com.horaapps.leafpic.Base.CustomAlbumsHandler.checkAndCreateAlbum(CustomAlbumsHandler.java:67)
W/System.err( 7582):    at com.horaapps.leafpic.Base.CustomAlbumsHandler.getSettings(CustomAlbumsHandler.java:48)
W/System.err( 7582):    at com.horaapps.leafpic.Base.Album.setSettings(Album.java:109)
W/System.err( 7582):    at com.horaapps.leafpic.Base.Album.<init>(Album.java:78)
W/System.err( 7582):    at com.horaapps.leafpic.PhotoPagerActivity.onCreate(PhotoPagerActivity.java:101)
E/AndroidRuntime( 7582): Process: com.horaapps.leafpic, PID: 7582
E/AndroidRuntime( 7582): java.lang.RuntimeException: Unable to resume activity {com.horaapps.leafpic/com.horaapps.leafpic.PhotoPagerActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.RelativeLayout.setBackgroundColor(int)' on a null object reference
E/AndroidRuntime( 7582):    at com.horaapps.leafpic.PhotoPagerActivity.setupUI(PhotoPagerActivity.java:216)
E/AndroidRuntime( 7582):    at com.horaapps.leafpic.PhotoPagerActivity.onResume(PhotoPagerActivity.java:251)
W/ActivityManager(  621):   Force finishing activity com.horaapps.leafpic/.PhotoPagerActivity
W/ActivityManager(  621): Activity pause timeout for ActivityRecord{1590eaf u0 com.horaapps.leafpic/.PhotoPagerActivity t30 f}
I/ActivityManager(  621): Process com.horaapps.leafpic (pid 7582) has died
I/PackageManager(  621): Setting last chosen activity com.horaapps.leafpic/.PhotoPagerActivity for user 0:
I/ActivityManager(  621): START u0 {act=android.intent.action.VIEW dat=content://media/external/images/media/11087 typ=image/gif flg=0x13000000 cmp=com.horaapps.leafpic/.PhotoPagerActivity (has extras)} from uid 10010 on display 0
I/ActivityManager(  621): Start proc 7952:com.horaapps.leafpic/u0a249 for activity com.horaapps.leafpic/.PhotoPagerActivity
W/System  ( 7952): ClassLoader referenced unknown path: /data/app/com.horaapps.leafpic-1/lib/arm
W/XPrivacy( 7952): Hooking package=com.horaapps.leafpic
W/System.err( 7952):    at com.horaapps.leafpic.utils.StringUtils.quoteReplace(StringUtils.java:157)
W/System.err( 7952):    at com.horaapps.leafpic.Base.CustomAlbumsHandler.checkAndCreateAlbum(CustomAlbumsHandler.java:67)
W/System.err( 7952):    at com.horaapps.leafpic.Base.CustomAlbumsHandler.getSettings(CustomAlbumsHandler.java:48)
W/System.err( 7952):    at com.horaapps.leafpic.Base.Album.setSettings(Album.java:109)
W/System.err( 7952):    at com.horaapps.leafpic.Base.Album.<init>(Album.java:78)
W/System.err( 7952):    at com.horaapps.leafpic.PhotoPagerActivity.onCreate(PhotoPagerActivity.java:101)
E/AndroidRuntime( 7952): Process: com.horaapps.leafpic, PID: 7952
E/AndroidRuntime( 7952): java.lang.RuntimeException: Unable to resume activity {com.horaapps.leafpic/com.horaapps.leafpic.PhotoPagerActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.RelativeLayout.setBackgroundColor(int)' on a null object reference
E/AndroidRuntime( 7952):    at com.horaapps.leafpic.PhotoPagerActivity.setupUI(PhotoPagerActivity.java:216)
E/AndroidRuntime( 7952):    at com.horaapps.leafpic.PhotoPagerActivity.onResume(PhotoPagerActivity.java:251)
W/ActivityManager(  621):   Force finishing activity com.horaapps.leafpic/.PhotoPagerActivity
W/ActivityManager(  621): Activity pause timeout for ActivityRecord{54e73c3 u0 com.horaapps.leafpic/.PhotoPagerActivity t32 f}
I/ActivityManager(  621): Process com.horaapps.leafpic (pid 7952) has died
I/ActivityManager(  621): START u0 {act=android.intent.action.VIEW dat=file:///storage/emulated/0/*******/******* Pictures/Dark.jpg typ=image/jpeg flg=0x10000001 cmp=com.horaapps.leafpic/.PhotoPagerActivity (has extras)} from uid 10093 on display 0
I/ActivityManager(  621): Start proc 8458:com.horaapps.leafpic/u0a249 for activity com.horaapps.leafpic/.PhotoPagerActivity
W/System  ( 8458): ClassLoader referenced unknown path: /data/app/com.horaapps.leafpic-1/lib/arm
W/XPrivacy( 8458): Hooking package=com.horaapps.leafpic
I/ActivityManager(  621): Displayed com.horaapps.leafpic/.PhotoPagerActivity: +976ms
I/PackageManager(  621): Setting last chosen activity com.horaapps.leafpic/.PhotoPagerActivity for user 0:
I/ActivityManager(  621): START u0 {act=android.intent.action.VIEW dat=content://ch.*******.app.fileprovider/tmp_files/****EXAMPLEFILE****.gif typ=image/* flg=0x3000001 cmp=com.horaapps.leafpic/.PhotoPagerActivity} from uid 10202 on display 0
W/System.err( 8458):    at com.horaapps.leafpic.utils.StringUtils.quoteReplace(StringUtils.java:157)
W/System.err( 8458):    at com.horaapps.leafpic.Base.CustomAlbumsHandler.checkAndCreateAlbum(CustomAlbumsHandler.java:67)
W/System.err( 8458):    at com.horaapps.leafpic.Base.CustomAlbumsHandler.getSettings(CustomAlbumsHandler.java:48)
W/System.err( 8458):    at com.horaapps.leafpic.Base.Album.setSettings(Album.java:109)
W/System.err( 8458):    at com.horaapps.leafpic.Base.Album.<init>(Album.java:78)
W/System.err( 8458):    at com.horaapps.leafpic.PhotoPagerActivity.onCreate(PhotoPagerActivity.java:101)
E/AndroidRuntime( 8458): Process: com.horaapps.leafpic, PID: 8458
E/AndroidRuntime( 8458): java.lang.RuntimeException: Unable to resume activity {com.horaapps.leafpic/com.horaapps.leafpic.PhotoPagerActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.RelativeLayout.setBackgroundColor(int)' on a null object reference
E/AndroidRuntime( 8458):    at com.horaapps.leafpic.PhotoPagerActivity.setupUI(PhotoPagerActivity.java:216)
E/AndroidRuntime( 8458):    at com.horaapps.leafpic.PhotoPagerActivity.onResume(PhotoPagerActivity.java:251)
W/ActivityManager(  621):   Force finishing activity com.horaapps.leafpic/.PhotoPagerActivity
W/ActivityManager(  621): Activity pause timeout for ActivityRecord{a8b6c9 u0 com.horaapps.leafpic/.PhotoPagerActivity t30 f}
I/ActivityManager(  621): Process com.horaapps.leafpic (pid 8458) has died
```

Note there are different exceptions...

Cyanogenmod 13 (based on Android 6.0.1)

**Comment by rugk** [comment]
I have a response from the developers of the other app (loosely translated):

> Surprisingly the NPE after an ACTION_VIEW intent also happens with file URIs.
> 
> You should not rely on Uri.getPath(). For file URIs this may work, but it will definitely fail with content URIs as these are only virtual paths. The only way to correctly open a content URI is to use the ContentResolver and methods such as openInputStream().

**Comment by craftyguy** [comment]
I also see this when opening a picture directly from Snap camera (camera in CyanogenMod 13) in leafpic

**Comment by rugk** [comment]
@craftyguy Interestingly I cannot reproduce it with the default camera app in CM13, but if it is the same error.
Would you mind also adding your logcat log? (Or at least compare if it is the same/similar)

**Comment by craftyguy** [comment]
@rugk

It's the same NPE as far as I can tell. Here's the logcat snippet:

```
05-06 07:14:52.779 W/System.err(5996): java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.lang.String.replace(java.lang.CharSequence, java.lang.CharSequence)' on a null object reference
05-06 07:14:52.786 W/System.err(5996):  at com.horaapps.leafpic.utils.StringUtils.quoteReplace(StringUtils.java:157)
05-06 07:14:52.786 W/System.err(5996):  at com.horaapps.leafpic.Base.CustomAlbumsHandler.checkAndCreateAlbum(CustomAlbumsHandler.java:67)
05-06 07:14:52.786 W/System.err(5996):  at com.horaapps.leafpic.Base.CustomAlbumsHandler.getSettings(CustomAlbumsHandler.java:48)
05-06 07:14:52.786 W/System.err(5996):  at com.horaapps.leafpic.Base.Album.setSettings(Album.java:109)
05-06 07:14:52.786 W/System.err(5996):  at com.horaapps.leafpic.Base.Album.<init>(Album.java:78)
05-06 07:14:52.786 W/System.err(5996):  at com.horaapps.leafpic.PhotoPagerActivity.onCreate(PhotoPagerActivity.java:101)
05-06 07:14:52.786 W/System.err(5996):  at android.app.Activity.performCreate(Activity.java:6251)
05-06 07:14:52.786 W/System.err(5996):  at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1108)
05-06 07:14:52.786 W/System.err(5996):  at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2403)
05-06 07:14:52.786 W/System.err(5996):  at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2510)
05-06 07:14:52.786 W/System.err(5996):  at android.app.ActivityThread.-wrap11(ActivityThread.java)
05-06 07:14:52.786 W/System.err(5996):  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1363)
05-06 07:14:52.786 W/System.err(5996):  at android.os.Handler.dispatchMessage(Handler.java:102)
05-06 07:14:52.786 W/System.err(5996):  at android.os.Looper.loop(Looper.java:148)
05-06 07:14:52.786 W/System.err(5996):  at android.app.ActivityThread.main(ActivityThread.java:5461)
05-06 07:14:52.786 W/System.err(5996):  at java.lang.reflect.Method.invoke(Native Method)
05-06 07:14:52.786 W/System.err(5996):  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
05-06 07:14:52.786 W/System.err(5996):  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
05-06 07:14:52.795 D/AndroidRuntime(5996): Shutting down VM
05-06 07:14:52.800 E/AndroidRuntime(5996): FATAL EXCEPTION: main
05-06 07:14:52.800 E/AndroidRuntime(5996): Process: com.horaapps.leafpic, PID: 5996
05-06 07:14:52.800 E/AndroidRuntime(5996): Theme: themes:{default=overlay:com.ciaostudio.ciaoui, iconPack:org.twelf.cmtheme, fontPkg:com.ciaostudio.ciaoui, com.android.systemui=overlay:com.ciaostudio.ciaoui, com.android.systemui.navbar=overlay:system}
05-06 07:14:52.800 E/AndroidRuntime(5996): java.lang.RuntimeException: Unable to resume activity {com.horaapps.leafpic/com.horaapps.leafpic.PhotoPagerActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.RelativeLayout.setBackgroundColor(int)' on a null object reference
05-06 07:14:52.800 E/AndroidRuntime(5996):  at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3137)
05-06 07:14:52.800 E/AndroidRuntime(5996):  at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3168)
05-06 07:14:52.800 E/AndroidRuntime(5996):  at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2515)
05-06 07:14:52.800 E/AndroidRuntime(5996):  at android.app.ActivityThread.-wrap11(ActivityThread.java)
05-06 07:14:52.800 E/AndroidRuntime(5996):  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1363)
05-06 07:14:52.800 E/AndroidRuntime(5996):  at android.os.Handler.dispatchMessage(Handler.java:102)
05-06 07:14:52.800 E/AndroidRuntime(5996):  at android.os.Looper.loop(Looper.java:148)
05-06 07:14:52.800 E/AndroidRuntime(5996):  at android.app.ActivityThread.main(ActivityThread.java:5461)
05-06 07:14:52.800 E/AndroidRuntime(5996):  at java.lang.reflect.Method.invoke(Native Method)
05-06 07:14:52.800 E/AndroidRuntime(5996):  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
05-06 07:14:52.800 E/AndroidRuntime(5996):  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
05-06 07:14:52.800 E/AndroidRuntime(5996): Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void android.widget.RelativeLayout.setBackgroundColor(int)' on a null object reference
05-06 07:14:52.800 E/AndroidRuntime(5996):  at com.horaapps.leafpic.PhotoPagerActivity.setupUI(PhotoPagerActivity.java:216)
05-06 07:14:52.800 E/AndroidRuntime(5996):  at com.horaapps.leafpic.PhotoPagerActivity.onResume(PhotoPagerActivity.java:251)
05-06 07:14:52.800 E/AndroidRuntime(5996):  at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1259)
05-06 07:14:52.800 E/AndroidRuntime(5996):  at android.app.Activity.performResume(Activity.java:6327)
05-06 07:14:52.800 E/AndroidRuntime(5996):  at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3126)
05-06 07:14:52.800 E/AndroidRuntime(5996):  ... 10 more
05-06 07:14:52.804 W/ActivityManager(3043):   Force finishing activity com.horaapps.leafpic/.PhotoPagerActivity
05-06 07:14:52.810 W/ActivityManager(3043):   Force finishing activity org.cyanogenmod.snap/com.android.camera.CameraLauncher


```

**Comment by dnldsht** [comment]
fixed this issue soon i will release v0.3 with a lot of improvements and bugs solved

EDIT: also hidden feature is now working!

**Comment by licaon-kter** [comment]
@DNLDsht 
Two things before release:
- pull translations
- tag the release so F-Droid can pick it up ;)

**Comment by dnldsht** [comment]
Sure just making some clean

**Comment by craftyguy** [comment]
I can confirm that this issue is resolved for me with the latest release (0.3), thanks for the quick fix!

**Comment by rugk** [comment]
I can confirm it too, BTW.
